tools/ci/templates/jobs-cpp.yaml

```diff
@@ -150,50 +151,62 @@ jobs:
       configuration: '${{parameters.buildConfig}}'
     timeoutInMinutes: 20
 
-  # Publish mrwebrtc.dll and associated files (PDB, ...)
+  # Stage artifacts
+  - task: CopyFiles@2
+    displayName: 'Stage mrwebrtc artifacts'
+    inputs:
+      sourceFolder: 'bin/${{parameters.buildPlatform}}/${{parameters.buildArch}}/${{parameters.buildConfig}}'
+      contents: '**/mrwebrtc.@(pdb|dll)'
+      targetFolder: '$(Build.ArtifactStagingDirectory)'
+
+  # Publish mrwebrtc.dll and mrwebrtc.pdb
   - task: PublishPipelineArtifact@0
     displayName: 'Publish mrwebrtc ($(buildTriple))'
     inputs:
       artifactName: 'mrwebrtc_$(buildTriple)'
-      targetPath: 'bin/${{parameters.buildPlatform}}/${{parameters.buildArch}}/${{parameters.buildConfig}}'
+      targetPath: '$(Build.ArtifactStagingDirectory)'
 
   # Delete PDBs to save disk space, otherwise the NuGet download below fails in Debug
-  - powershell: |
-      Remove-Item "bin/${{parameters.buildPlatform}}/${{parameters.buildArch}}/${{parameters.buildConfig}}/*" -Force -Include "*.pdb" -Exclude "Microsoft.MixedReality.WebRTC.*","NativeLib.*","NativeTests.*"
-    displayName: 'Delete PDBs after linking webrtc.lib'
-    condition: eq(variables['MRWebRTC_RestorePDBs'], 'true')
+  - ${{ if eq(parameters.withTesting, 'true') }}:
+    - powershell: |
+        Remove-Item "bin/${{parameters.buildPlatform}}/${{parameters.buildArch}}/${{parameters.buildConfig}}/*" -Force -Include "*.pdb" -Exclude "Microsoft.MixedReality.WebRTC.*","NativeLib.*","NativeTests.*"
+      displayName: 'Delete PDBs after linking webrtc.lib'
+      condition: eq(variables['MRWebRTC_RestorePDBs'], 'true')
 
   # Restore the NuGet packages for the mrwebrtc tests
-  - task: 333b11bd-d341-40d9-afcf-b32d5ce6f23b@2  # NuGetCommand@2
-    displayName: 'NuGet restore tests'
-    inputs:
-      command: restore
-      restoreSolution: '$(Build.SourcesDirectory)/tools/build/mrwebrtc/win32/tests/packages.config'
-      restoreDirectory: '$(Build.SourcesDirectory)/packages'
-      includeNuGetOrg: true
-      feedsToUse: 'config'
-      nugetConfigPath: '$(NuGetConfigPath)'
-    timeoutInMinutes: 10
+  - ${{ if eq(parameters.withTesting, 'true') }}:
```

Comment:
nit: move the testing steps into a template and include it from here, so you don't need to check `withTesting` 4 times

Reply:
Let's leave as TODO, there's much cleaning to be done on pipelines.